[RDY] Use C++11 standard threading for th_movie.cpp

TheCycoONE opened the pull request:
Replace SDL threading with C++11 threading.

In making this change I already fixed one possible deadlock where the mutex was
not released in write if the picture didn't have a buffer.

TheCycoONE commented:
Sorry for the noise, I keep changing my mind on how far I want to take this. There is one more modification I want to make.

Alberth289346 commented on CorsixTH/Src/th_movie.cpp:
```diff
@@ -166,24 +145,26 @@ void THMoviePictureBuffer::allocate(SDL_Renderer *pRenderer, int iWidth, int iHe
     for(int i = 0; i < ms_pictureBufferSize; i++)
     {
-        SDL_LockMutex(m_aPictureQueue[i].m_pMutex);
+        std::lock_guard<std::mutex> pictureLock(m_aPictureQueue->m_mutex);
```
Weird that you don't need a [i] here, while the previous version had one, and line 256 also seems to have one

TheCycoONE replied:
Oops, yes this looks wrong.

Alberth289346 commented on CorsixTH/Src/th_movie.cpp:
```diff
@@ -136,9 +116,8 @@ THMoviePictureBuffer::~THMoviePictureBuffer()
-    SDL_CondSignal(m_pCond);
-    SDL_UnlockMutex(m_pMutex);
+    std::lock_guard<std::mutex> lock(m_mutex);
+    m_cond.notify_all();
```
SDL woke up 1 thread, but  I guess this is fine, as everybody will find the m_fAborting flag, and quit?

TheCycoONE replied:
Intentional - releasing should unlock every thread waiting for the queue. It's theoretical though because currently only one other thread would be waiting.

TheCycoONE commented:
Fixed the picture lock, and added a second commit to make the aborting flag atomic, since it was relying on unsafe / undefined behavior (there is no guarantee without atomic that other threads would ever see the updated value, or that the code wouldn't be reordered.)
